macros/check-fips

```diff
@@ -68,6 +66,79 @@ for b in $(find "${BUILDROOT_BINDIR}" "${BUILDROOT_LIBEXECDIR}" -type f -executa
       (( not_found+=1 ))
     fi
   fi
+}
+
+is_rust_bin() {
+  local bin output
+  bin="${1:?}"
+  output="$("${STRINGS}" "${bin}")"
+  grep -Fq -m1 -e 'rust_alloc' -e 'rust_begin' <<<"${output}"
+}
+
+uses_ring() {
+  local bin output
+  bin="${1:?}"
+  output="$("${STRINGS}" "${bin}")"
+  grep -Fq -m1 '/ring' <<<"${output}"
+}
+
+uses_aws_lc_rs() {
+  local bin output
+  bin="${1:?}"
+  output="$("${STRINGS}" "${bin}")"
+  grep -Fq -m1 'aws-lc-rs' <<<"${output}"
+}
+
+uses_aws_lc_fips_sys() {
+  local bin output
+  bin="${1:?}"
+  output="$("${STRINGS}" "${bin}")"
+  grep -Fq -m1 'aws-lc-fips-sys' <<<"${output}"
+}
+
+check_rust_bin() {
+  local b f
+  b="${1:?}"
+  f="${2:?}"
+
+  if uses_ring "${b}" && ! uses_aws_lc_rs "${b}"; then
+    echo "${b} must use aws-lc-rs" >&2
+    (( miscompiled+=1 ))
+  fi
```

Comment:
When would binaries use both? That seems like an error also.

To cut down on the repeated `strings` calls on the same binaries, I would put this check at the end of the next one:
```
if uses_aws_lc_rs "${b}"; then
...
else
  # doesn't use aws_lc_rs
  if uses_ring "${b}"; then
    echo "${b} must use aws-lc-rs" >&2
    (( miscompiled+=1 ))
  fi
fi